fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -57,6 +57,12 @@
 public class ResourceDAOImpl extends FHIRDbDAOImpl implements ResourceDAO {
     private static final Logger log = Logger.getLogger(ResourceDAOImpl.class.getName());
     private static final String CLASSNAME = ResourceDAOImpl.class.getName();
+
+    // Per issue with private memory in db2, we have set this to 1M.
+    // Anything larger than 1M is then inserted into the db with an update.
+    private static final int SIZE_LIMIT = 1048576;
+    private static final String LARGE_BLOB = "UPDATE %s.%s_RESOURCES SET data = ? WHERE logical_resource_id = ?";
```

Comment:
That's going to overwrite the data for every version. You need to use `resource_id = ?` which is the PK for this table.

Reply:
Right - great point, fixed